v8/upgrade_metadata_v8/upgrade_metadata_v8.py

```diff
@@ -0,0 +1,182 @@
+            for post in flagged:
+                L.info('    ' + (post.metadata_path if post.is_two_file else post.source_path))
+            L.warn('Please make a backup before running this plugin. It might eat your data.')
+            if not options['yes']:
+                yesno = utils.ask_yesno("Proceed with metadata upgrade?")
+            if options['yes'] or yesno:
+                no_converted = 0
+                no_converted_partial = 0
+                for post in flagged:
+                    converted = False
+                    fully_converted = True
+                    for lang in self.site.config['TRANSLATIONS'].keys():
+                        # Get file names and extractor
+                        extractor = post.used_extractor[lang]
+                        is_two_file = post.is_two_file
+                        if lang == post.default_lang:
+                            fname = post.metadata_path if is_two_file else post.source_path
+                        else:
+                            meta_path = os.path.splitext(post.source_path)[0] + '.meta' if is_two_file else post.source_path
+                            fname = utils.get_translation_candidate(post.config, meta_path, lang)
+
+                        # We don't handle compilers which extract metadata for now
+                        if post.compiler is extractor:
+                            L.warn('Cannot convert {0} (language {1}), as metadata was extracted by compiler.'.format(fname, lang))
+                            fully_converted = False
+                            continue
+
+                        # Read metadata and text from post file
+                        with io.open(fname, "r", encoding="utf-8-sig") as meta_file:
+                            source_text = meta_file.read()
+                        if not is_two_file:
+                            _, content_str = extractor.split_metadata_from_text(source_text)
+                        meta = extractor.extract_text(source_text)
+
+                        # Consider metadata mappings
+                        sources = {
+                            'tags': 'tags',
+                            'status': 'status',
+                            'has_math': 'has_math',
+                        }
+                        for foreign, ours in self.site.config.get('METADATA_MAPPING', {}).get(extractor.map_from, {}).items():
+                            if ours in sources:
+                                sources[ours] = foreign
+                        for meta_key, hook in self.site.config.get('METADATA_VALUE_MAPPING', {}).get(extractor.map_from, {}).items():
+                            if meta_key in sources.values():
+                                L.warn('Cannot convert {0} (language {1}): a metadata value mapping is defined for "{2}"!'.format(fname, lang, meta_key))
+
+                        # Update metadata
+                        updated = False
```

Comment:
Feature request:

if `section` is present, but `category` isn’t: set `category` to `section` value and `remove` section  
if both are present: show a warning (as is done on the `nuke-sections` branch of the main repo)

Reply:
Implemented in 685be9cb24cf13a4b9978baefa2e01c44ef2200d.